### ISSUE TITLE
Pyinfra errors with cryptic message when SFTP is not avilable

### ISSUE THREAD

**Issue by relaxdiego** [Fizzadar/pyinfra#499]
**Describe the bug**
When attempting to upload a file to a remote server and the remote server happens to have SFTP disabled (e.g. Synology NAS), Pyinfra simply errors out with a cryptic message:

```
[192.168.86.43] Command socket/SSH error: SSHException('Channel closed.',)
[192.168.86.43] Error
```

**To Reproduce**

Perform the following operation on any remote that has [the SFTP subsystem disabled](https://devanswers.co/ubuntu-disable-sftp/):

```python
files.put(
    name="Put some file in",
    src='/path/in/local',
    dest='/path/in/remote',
)
```

**Expected behavior**

Pyinfra should provide a more helpful error message so that the user can immediately fix the issue on their part.

**Meta**
+ Include output of `pyinfra --support`.

```
System: Darwin
  Platform: Darwin-19.6.0-x86_64-i386-64bit
  Release: 19.6.0
  Machine: x86_64
pyinfra: v1.3
Executable: /Users/mark/.pyenv/versions/aircraft/bin/pyinfra
Python: 3.7.7 (CPython, Clang 11.0.0 (clang-1100.0.33.17))
```

+ How was pyinfra installed (source/pip)?

```
pip install -e .
```

**Possible Fix**

```diff
diff --git a/pyinfra/api/connectors/ssh.py b/pyinfra/api/connectors/ssh.py
index 7a28b63a..926107ba 100644
--- a/pyinfra/api/connectors/ssh.py
+++ b/pyinfra/api/connectors/ssh.py
@@ -306,7 +306,14 @@ def run_shell_command(
 @memoize
 def _get_sftp_connection(host):
     transport = host.connection.get_transport()
-    return SFTPClient.from_transport(transport)
+
+    try:
+        sftp_client = SFTPClient.from_transport(transport)
+    except SSHException as e:
+        raise Exception("Unable to establish SFTP connection. "
+                        "Check that the SFTP subsystem "
+                        f"for the SSH service at {host} is enabled.") from e
+    return sftp_client
 
 
 def _get_file(host, remote_filename, filename_or_io):
```

**Comment by Fizzadar** [comment]
Thank for the PR/patch @relaxdiego, change looks good to me I shall get this added over the next few days!

**Comment by Fizzadar** [comment]
Now implemented in https://github.com/Fizzadar/pyinfra/commit/c4160325a7e12a3048a9727940c46db84a9975d7 pending `v1.3.1` release!

**Comment by Fizzadar** [comment]
Now released in `1.3.1`!
